### ISSUE TITLE
Bug with ExpressionCompiler in 1.1

### ISSUE THREAD

**Issue by stsrki** [Megabit/Blazorise#4137]
Once I tried to publish demo apps, I got this error.

![image](https://user-images.githubusercontent.com/900302/192531015-71c75299-76cf-4150-a9d7-a0c4fc93fa28.png)

Further investigation showed me that `instance.GetType()` in `ExpressionCompiler.CreatePropertyGetter` is behaving differently:

**WebAssembly demo apps**

![image](https://user-images.githubusercontent.com/900302/192531248-58322a7a-1c70-4753-9c7b-eade250ba37f.png)

**Server demo apps**

![image](https://user-images.githubusercontent.com/900302/192531403-6398e34b-13ad-4d08-9ae9-febc7f5b7122.png)

---

I have added try/catch as a workaround, but we need to fix this ASAP.

```cs
private IList<object> LoadServiceProviderDisposableList()
{
    try
    {
        if ( disposablesGetter is null )
            disposablesGetter = ExpressionCompiler.CreatePropertyGetter<IList<object>>( ServiceProvider, PROPERTY_DISPOSABLES );

        var disposables = disposablesGetter( ServiceProvider );

        disposePossible = !disposables.IsNullOrEmpty();

        return disposables;
    }
    catch
    {
        disposePossible = false;
    }

    return new List<object>();
}
```

**Comment by David-Moreira** [comment]
Lol.. that's new... what kind of dark magic is this?

**Comment by David-Moreira** [comment]
Eitherway after some testing.... Even the previous straight up reflection version does nothing at all. I still have no idea how to solve this. And I'd prefer a WASM check then a try/catch in the meantime?

I was doing some investigation and I found this:
Maybe this is what should be used?
![image](https://user-images.githubusercontent.com/22283161/192633165-47ffc2e9-779d-445e-9728-1d215acbef5e.png)

https://learn.microsoft.com/en-us/dotnet/api/microsoft.aspnetcore.components.owningcomponentbase?view=aspnetcore-7.0

**Comment by paulsterling** [comment]
As suggested by @stsrki I this class to override IComponentDisposer, then registered for DI. In case this is useful for anyone else.

```
using Blazorise;

namespace Mis.Mw.App.Client
{
    public class BlazoriseWorkaround : IComponentDisposer
    {
        void IComponentDisposer.Dispose<TComponent>(TComponent component)
        {
            // nothing to see here
        }
    }
}
```

Program.cs
```
            builder.Services
              .AddBlazorise(options =>
              {
                  options.Immediate = true;
              })
              .AddBlazoriseRichTextEdit()
              .AddBootstrapComponents()
              .AddBootstrapProviders()
              .AddFontAwesomeIcons();

            builder.Services.AddScoped<IComponentDisposer, BlazoriseWorkaround>();
```

**Comment by stsrki** [comment]
Fixed in 1.1.2
